tests/net/buf/src/main.c

```diff
@@ -724,4 +725,60 @@ ZTEST(net_buf_tests, test_net_buf_user_data)
 	net_buf_unref(buf);
 }
 
+ZTEST(net_buf_tests, test_net_buf_comparison)
+{
+	struct net_buf *buf;
+	size_t written;
+	size_t offset;
+	size_t to_compare;
+	size_t res;
+	uint8_t data[FIXED_BUFFER_SIZE * 2];
+
+	/* Fill data buffer */
+	for (int i = 0; i < sizeof(data); ++i) {
+		data[i] = (uint8_t)i;
+	}
+
+	/* Allocate a single net_buf  */
+	buf = net_buf_alloc(&fixed_pool, K_NO_WAIT);
+	zassert_not_null(buf, "Failed to get buffer");
+
+	written = net_buf_append_bytes(buf, buf->size, data, K_NO_WAIT, NULL, NULL);
+	zassert_equal(written, buf->size, "Failed to fill the buffer");
+	zassert_equal(buf->frags, NULL, "Additional buffer allocated");
+
```

Comment:
Could we also add a test that should fail (negative testing), now all the tests verify that the function passes ok.